[pickers] Clean slots on `PickersArrowSwitcher` component (#5890)
mui/mui-x: docs/data/date-pickers/custom-components/custom-components.md

In the examples below we provide ways of implementing date range shortcuts using this slot.

{{"demo": "PaperContentComponent.js", "defaultCodeOpen": false, "disableAd": true}}

## Arrow switcher buttons and icons

:::info

> Slots: `LeftArrowIcon` / `RightArrowIcon` / `LeftArrowButton` / `RightArrowButton`

Author: @samuelsycamore @joserodolfofreitas I would be interested in your feedback here
I am trying to list the slots that the section is talking about

I would like to apply the same UI on every section of this slots documentation.

Reviewer: Interesting. We should definitely give this some thought. My first thought would be to create a list, but after seeing this text on the page I appreciate how concise it is. Maybe we should treat it like a plain sentence and use commas instead?

```suggestion
Slots: `LeftArrowIcon`, `RightArrowIcon`, `LeftArrowButton`, `RightArrowButton`.
```

Author: We will have a similar topics for sections concerning several picker components but not all

For instance:

```
Components supporting this behavior: DatePicker, DateTimePicker, DateRangePicker, DateTimeRangePicker
```

Reviewer: Maybe a list would be better after all? It would be easier to quickly scan.

Author: I wonder if we could create some custom style (a bit like the `:::info`) that lets you know that this small block contains information about the scope of the section

Reviewer: That's an interesting idea @flaviendelangle. Maybe @danilo-leal or @gerdadesign have thoughts about this?

Reviewer 2: Not sure I'm following — can you provide more context?

Author: Sorry if I explains things that are already obvious to you:

In our components, we have *slots* which are parts of UI that users can customize by passing props to them or override with there own component.
For instance `<DataGrid components={{ Toolbar: MyToolbar }} />`, `<DatePicker components={{ Input: JoyTextField }} />` (not implemented yet for the 2nd one).

In the doc, we have:
- a `Slots` section in the API page that lists the slots ([here][https://mui.com/x/api/date-pickers/date-picker/#slots)).

- a `Custom components` ([data grid][https://mui.com/x/react-data-grid/components/), [pickers](https://mui.com/x/react-date-pickers/custom-components/)) that gives detailed example about how to use the slots. Each section of these pages can cover one or several slots (for instance in this PR we have 4 slots that works together so we document them in the same section).

My main goal was to provide a link between the two.
I would like to be able to correctly document the props of the slots on the API section.
And from the `Custom components` pages to be able to access those props by clicking the name of the slot.

:::

> The following slots allows you to customize how to render the buttons and icons for an arrow switcher component (the component to navigate to the "Previous" and "Next" step of the picker)

Reviewer 3: @samuelsycamore Do we have conventions on [which dash/hyphen](https://jakubmarian.com/hyphen-minus-en-dash-and-em-dash-difference-and-usage-in-english/) to use when?
As far as I can see, there are only a few instances of other than the minus dash in codebase... 🤔

In the example below, the tabs are using different icons and have an additional component:

{{"demo": "Tabs.js", "disableAd": true}}

## Arrow switcher buttons and icons

:::info
Slots:

- `LeftArrowIcon`
- `RightArrowIcon`
- `LeftArrowButton`
- `RightArrowButton`

> :::

Reviewer: Agreed @alexfauquette, it creates a lot of dead space to the right side of the list. Does this need to be a callout at all? What if we just listed them directly in the text? (Suggestion below.)

> :::

Author: My goal would be at some point to document the slot props correctly in the API section and to link from here (or from the bottom of the page if we split the page into several pages)

> The following slots let you customize how to render the buttons and icons for an arrow switcher component—the component to navigate to the "Previous" and "Next" steps of the picker.

Reviewer: ```suggestion
The following slots let you customize how to render the buttons and icons for an arrow switcher component—the component to navigate to the "Previous" and "Next" steps of the picker: `LeftArrowIcon`, `RightArrowIcon`, `LeftArrowButton`, `RightArrowButton`.
```

### Component props

> You can pass props to the icon and buttons as show below:

Reviewer: ```suggestion
You can pass props to the icons and buttons as shown below:
```

{{"demo": "ArrowSwitcherComponentProps.js", "defaultCodeOpen": false, "disableAd": true}}

### Component

> You can pass custom components, for example to replace the icons as show below:

Reviewer: ```suggestion
You can pass custom components—to replace the icons, for example—as shown below:
```
